artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -1434,6 +1436,10 @@ private boolean canDispatch() {
             return false;
          }
 
+         if (isExclusive() && hasExclusivePendingDeliveries()) {
+            return false;
+         }
```

Comment:
Doesnt this mean it will only send 1 outstanding message at a time? That doesnt seem desirable?

Reply:
(Sometimes maybe, but overall no due to the earlier-and-later 'remember can dispatch' logic I had overlooked)